Makefile permissions issue fix

snowskeleton opened the pull request:
Adds `sudo` to the lines `mv .build/release/vapor ${DEST}` and `rm ${DEST}`, since root level permissions are required to make changes to `$DEST` (/usr/local/bin). (Note that simply running `sudo make install` without this change will result in errors.) Also added a line to change the permissions of the build moved into `/usr/local/bin` to match with standard conventions.
Possibly fixes issue #347, but not enough detail is noted in said issue to know if it's the same problem.

siemensikkema reviewed (commented):
Makes a lot of sense to me! Is it common practice to have `sudo` _inside_ `Makefile`s as opposed to invoking `make` itself using `sudo`?

snowskeleton commented:
Running `sudo make install` results in errors, since it tries to apply `sudo` to everything being run rather than just the `mv` and `rm` commands.
As for whether or not it’s common practice, I have no clue, since I’ve never worked with `make` before. I did a bit of searching to see if I was breaking any rules, and I couldn’t find any posts specifically warning against it.

siemensikkema reviewed (approved):
LGTM!

siemensikkema commented:
@snowskeleton could you update the title and body of the PR with a concise description of the fix for the release? See https://github.com/vapor/vapor/blob/master/.github/contributing.md

snowskeleton commented:
@siemensikkema title and comments updated. How does it look now?

siemensikkema commented:
@snowskeleton for a second I thought that was a clash between the `.build` folder and the `build` executable but that's not the case. The thing is that I cannot reproduce the problem ... @0xTim what was the problem you were experiencing in #347 ?

snowskeleton commented:
@siemensikkema here is the output when I run `make install`. This is on Ubuntu 20.04.1 LTS.
```
snow@ziggy:~/toolbox$ make install
swiftc ./scripts/build.swift
./scripts/build.swift:48:10: warning: 'launchPath' is deprecated: renamed to 'executableURL'
    task.launchPath = "/usr/bin/env"
         ^
./scripts/build.swift:48:10: note: use 'executableURL' instead
    task.launchPath = "/usr/bin/env"
         ^~~~~~~~~~
         executableURL
./scripts/build.swift:50:10: warning: 'launch()' is deprecated: renamed to 'run'
    task.launch()
         ^
./scripts/build.swift:50:10: note: use 'run' instead
    task.launch()
         ^~~~~~
         run
./scripts/build.swift:61:10: warning: 'launchPath' is deprecated: renamed to 'executableURL'
    task.launchPath = "/usr/bin/env"
         ^
./scripts/build.swift:61:10: note: use 'executableURL' instead
    task.launchPath = "/usr/bin/env"
         ^~~~~~~~~~
         executableURL
./scripts/build.swift:69:10: warning: 'launch()' is deprecated: renamed to 'run'
    task.launch()
         ^
./scripts/build.swift:69:10: note: use 'run' instead
    task.launch()
         ^~~~~~
         run
./build
$ swift build --disable-sandbox --configuration release -Xswiftc -cross-module-optimization --enable-test-discovery
[2/2] Linking vapor
rm ./build
mv .build/release/vapor /usr/local/bin/vapor
mv: cannot move '.build/release/vapor' to '/usr/local/bin/vapor': Permission denied
make: *** [Makefile:8: install] Error 1
```
It's specifically that penultimate line that I was trying to fix (`mv: cannot move '.build/release/vapor' to '/usr/local/bin/vapor': Permission denied`).

snowskeleton commented:
I just tested on MacOS (10.15.7) and it ran just fine (I made sure to clear `sudo` cache as prior to `make install` as well). Looking at the file permissions, the convention is different on MacOS versus Linux, it seems, so `sudo` isn't needed.

0xTim commented:
@siemensikkema yeah the issue is when trying to install it on Linux

0xTim commented:
Just checked and this won't clash with the Homebrew installation

0xTim reviewed (approved):
Thanks!
